returnn/datasets/concat_files.py

```diff
@@ -103,13 +143,13 @@ def __init__(
         self.preload_next_n_sub_epochs = preload_next_n_sub_epochs
         self.buffer_size = buffer_size
         self.file_cache_opts = file_cache_opts or {}
-        self._file_sizes: Optional[Dict[str, int]] = None  # file -> size. for equal distribution across sub epochs
+        self._file_sizes: Optional[Dict[str, int]] = None  # key -> size. for equal distribution across sub epochs
```

Comment:
Maybe use the index (`int`) in the `files` list instead?
```suggestion
        self._file_sizes: Optional[Dict[int, int]] = None  # file index -> size. for equal distribution across sub epochs
```

Reply:
I was doing this but then we have to be really careful with shuffling the files list and making sure we don‘t mess up any indices. The ad hoc key circumvents this.

Reply:
Hm right. Ok, as you prefer. But maybe make a separate function for it (`_key_for_files` or so).